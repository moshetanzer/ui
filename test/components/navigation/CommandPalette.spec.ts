import { describe, it, expect } from 'vitest'
import { UCommandPalette } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('CommandPalette', () => {
  it.each([
    ['with placeholder', { props: { placeholder: 'Emptystate placeholder' } }],
    ['with multiple', { props: { multiple: true } }],
    ['with nullable', { props: { nullable: true } }],
    ['with searchable', { props: { icon: 'i-heroicons-academic-cap', searchable: false } }],
    ['with loading', { props: { loading: true } }],
    ['with groups', { props: { groups: [{ key: '1', label: 'Group 1', commands: [{ label: 'Command 1' }] }] } }],
    ['with icon', { props: { icon: 'i-heroicons-arrow-up-on-square-stack' } }],
    ['with loading icon', { props: { loading: true, loadingIcon: 'i-heroicons-arrows-up-down-solid' } }],
    ['with icon', { props: { icon: 'i-heroicons-arrow-up-on-square-stack' } }],
    ['with selected icon', { props: { selectedIcon: 'i-heroicons-check', groups: [{ key: '1', label: 'Group 1', commands: [{ label: 'Command 1' }] }] } }],
    ['with close button', { props: { closeButton: { label: 'close button' } } }],
    ['with emptystate', { props: { emptyState: { icon: 'i-heroicons-academic-cap', label: 'Emptystate label', queryLabel: 'Emptystate queryLabel' } } }]
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof UCommandPalette.props>) => {
    const html = await ComponentRender(nameOrHtml, options, UCommandPalette)
    expect(html).toMatchSnapshot()
  })
})
