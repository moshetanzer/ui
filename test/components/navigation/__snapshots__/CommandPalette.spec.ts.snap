// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CommandPalette > renders with close button correctly 1`] = `
"<div class="flex flex-col flex-1 min-h-0 divide-y divide-gray-100 dark:divide-gray-800" data-headlessui-state="">
  <div class="relative flex items-center"><span class="i-heroicons-magnifying-glass-20-solid pointer-events-none absolute start-4 text-gray-400 dark:text-gray-500 h-5 w-5" aria-hidden="true"></span><input value="" id="headlessui-combobox-input-14" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="w-full placeholder-gray-400 dark:placeholder-gray-500 bg-transparent border-0 text-gray-900 dark:text-white focus:ring-0 focus:outline-none sm:text-sm h-12 px-4 ps-11 pe-10" placeholder="Search..." aria-label="Search..." autocomplete="off"><button type="button" class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center absolute end-4" aria-label="Close">
      <!--v-if--><span class="">close button</span>
      <!--v-if-->
    </button></div>
  <div class="flex flex-col items-center justify-center flex-1 px-6 py-14 sm:px-14"><span class="i-heroicons-magnifying-glass-20-solid w-6 h-6 mx-auto text-gray-400 dark:text-gray-500 mb-4" aria-hidden="true"></span>
    <p class="text-sm text-center text-gray-900 dark:text-white">We couldn't find any items with that term. Please try again.</p>
  </div>
</div>"
`;

exports[`CommandPalette > renders with emptystate correctly 1`] = `
"<div class="flex flex-col flex-1 min-h-0 divide-y divide-gray-100 dark:divide-gray-800" data-headlessui-state="">
  <div class="relative flex items-center"><span class="i-heroicons-magnifying-glass-20-solid pointer-events-none absolute start-4 text-gray-400 dark:text-gray-500 h-5 w-5" aria-hidden="true"></span><input value="" id="headlessui-combobox-input-15" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="w-full placeholder-gray-400 dark:placeholder-gray-500 bg-transparent border-0 text-gray-900 dark:text-white focus:ring-0 focus:outline-none sm:text-sm h-12 px-4 ps-11" placeholder="Search..." aria-label="Search..." autocomplete="off">
    <!--v-if-->
  </div>
  <div class="flex flex-col items-center justify-center flex-1 px-6 py-14 sm:px-14"><span class="i-heroicons-academic-cap w-6 h-6 mx-auto text-gray-400 dark:text-gray-500 mb-4" aria-hidden="true"></span>
    <p class="text-sm text-center text-gray-900 dark:text-white">Emptystate queryLabel</p>
  </div>
</div>"
`;

exports[`CommandPalette > renders with groups correctly 1`] = `
"<div class="flex flex-col flex-1 min-h-0 divide-y divide-gray-100 dark:divide-gray-800" data-headlessui-state="">
  <div class="relative flex items-center"><span class="i-heroicons-magnifying-glass-20-solid pointer-events-none absolute start-4 text-gray-400 dark:text-gray-500 h-5 w-5" aria-hidden="true"></span><input value="" id="headlessui-combobox-input-5" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="w-full placeholder-gray-400 dark:placeholder-gray-500 bg-transparent border-0 text-gray-900 dark:text-white focus:ring-0 focus:outline-none sm:text-sm h-12 px-4 ps-11" placeholder="Search..." aria-label="Search..." autocomplete="off" aria-controls="headlessui-combobox-options-6" aria-activedescendant="headlessui-combobox-option-7">
    <!--v-if-->
  </div>
  <div id="headlessui-combobox-options-6" role="listbox" data-headlessui-state="" aria-label="Commands" class="relative flex-1 overflow-y-auto divide-y divide-gray-100 dark:divide-gray-800 scroll-py-2">
    <div class="p-2" role="option">
      <h2 class="px-2.5 my-2 text-xs font-semibold text-gray-900 dark:text-white">Group 1</h2>
      <div class="text-sm text-gray-700 dark:text-gray-200" role="listbox" aria-label="Group 1">
        <div id="headlessui-combobox-option-7" role="option" aria-selected="true" class="flex justify-between select-none items-center rounded-md px-2.5 py-1.5 gap-2 relative bg-gray-100 dark:bg-gray-800 text-gray-900 dark:text-white cursor-pointer" data-headlessui-state="active selected">
          <div class="flex items-center gap-1.5 min-w-0">
            <!--v-if-->
            <div class="flex items-center gap-1.5 min-w-0">
              <!--v-if--><span class="truncate">Command 1</span><!-- eslint-disable-next-line vue/no-v-html -->
              <!--v-if-->
            </div>
          </div><span class="i-heroicons-check-20-solid h-5 w-5 text-gray-900 dark:text-white flex-shrink-0" aria-hidden="true"></span>
        </div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`CommandPalette > renders with icon correctly 1`] = `
"<div class="flex flex-col flex-1 min-h-0 divide-y divide-gray-100 dark:divide-gray-800" data-headlessui-state="">
  <div class="relative flex items-center"><span class="i-heroicons-arrow-up-on-square-stack pointer-events-none absolute start-4 text-gray-400 dark:text-gray-500 h-5 w-5" aria-hidden="true"></span><input value="" id="headlessui-combobox-input-8" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="w-full placeholder-gray-400 dark:placeholder-gray-500 bg-transparent border-0 text-gray-900 dark:text-white focus:ring-0 focus:outline-none sm:text-sm h-12 px-4 ps-11" placeholder="Search..." aria-label="Search..." autocomplete="off">
    <!--v-if-->
  </div>
  <div class="flex flex-col items-center justify-center flex-1 px-6 py-14 sm:px-14"><span class="i-heroicons-magnifying-glass-20-solid w-6 h-6 mx-auto text-gray-400 dark:text-gray-500 mb-4" aria-hidden="true"></span>
    <p class="text-sm text-center text-gray-900 dark:text-white">We couldn't find any items with that term. Please try again.</p>
  </div>
</div>"
`;

exports[`CommandPalette > renders with icon correctly 2`] = `
"<div class="flex flex-col flex-1 min-h-0 divide-y divide-gray-100 dark:divide-gray-800" data-headlessui-state="">
  <div class="relative flex items-center"><span class="i-heroicons-arrow-up-on-square-stack pointer-events-none absolute start-4 text-gray-400 dark:text-gray-500 h-5 w-5" aria-hidden="true"></span><input value="" id="headlessui-combobox-input-10" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="w-full placeholder-gray-400 dark:placeholder-gray-500 bg-transparent border-0 text-gray-900 dark:text-white focus:ring-0 focus:outline-none sm:text-sm h-12 px-4 ps-11" placeholder="Search..." aria-label="Search..." autocomplete="off">
    <!--v-if-->
  </div>
  <div class="flex flex-col items-center justify-center flex-1 px-6 py-14 sm:px-14"><span class="i-heroicons-magnifying-glass-20-solid w-6 h-6 mx-auto text-gray-400 dark:text-gray-500 mb-4" aria-hidden="true"></span>
    <p class="text-sm text-center text-gray-900 dark:text-white">We couldn't find any items with that term. Please try again.</p>
  </div>
</div>"
`;

exports[`CommandPalette > renders with loading correctly 1`] = `
"<div class="flex flex-col flex-1 min-h-0 divide-y divide-gray-100 dark:divide-gray-800" data-headlessui-state="">
  <div class="relative flex items-center"><span class="i-heroicons-arrow-path-20-solid pointer-events-none absolute start-4 text-gray-400 dark:text-gray-500 h-5 w-5 animate-spin" aria-hidden="true"></span><input value="" id="headlessui-combobox-input-4" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="w-full placeholder-gray-400 dark:placeholder-gray-500 bg-transparent border-0 text-gray-900 dark:text-white focus:ring-0 focus:outline-none sm:text-sm h-12 px-4 ps-11" placeholder="Search..." aria-label="Search..." autocomplete="off">
    <!--v-if-->
  </div>
  <div class="flex flex-col items-center justify-center flex-1 px-6 py-14 sm:px-14"><span class="i-heroicons-magnifying-glass-20-solid w-6 h-6 mx-auto text-gray-400 dark:text-gray-500 mb-4" aria-hidden="true"></span>
    <p class="text-sm text-center text-gray-900 dark:text-white">We couldn't find any items with that term. Please try again.</p>
  </div>
</div>"
`;

exports[`CommandPalette > renders with loading icon correctly 1`] = `
"<div class="flex flex-col flex-1 min-h-0 divide-y divide-gray-100 dark:divide-gray-800" data-headlessui-state="">
  <div class="relative flex items-center"><span class="i-heroicons-arrows-up-down-solid pointer-events-none absolute start-4 text-gray-400 dark:text-gray-500 h-5 w-5 animate-spin" aria-hidden="true"></span><input value="" id="headlessui-combobox-input-9" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="w-full placeholder-gray-400 dark:placeholder-gray-500 bg-transparent border-0 text-gray-900 dark:text-white focus:ring-0 focus:outline-none sm:text-sm h-12 px-4 ps-11" placeholder="Search..." aria-label="Search..." autocomplete="off">
    <!--v-if-->
  </div>
  <div class="flex flex-col items-center justify-center flex-1 px-6 py-14 sm:px-14"><span class="i-heroicons-magnifying-glass-20-solid w-6 h-6 mx-auto text-gray-400 dark:text-gray-500 mb-4" aria-hidden="true"></span>
    <p class="text-sm text-center text-gray-900 dark:text-white">We couldn't find any items with that term. Please try again.</p>
  </div>
</div>"
`;

exports[`CommandPalette > renders with multiple correctly 1`] = `
"<div class="flex flex-col flex-1 min-h-0 divide-y divide-gray-100 dark:divide-gray-800" data-headlessui-state="">
  <div class="relative flex items-center"><span class="i-heroicons-magnifying-glass-20-solid pointer-events-none absolute start-4 text-gray-400 dark:text-gray-500 h-5 w-5" aria-hidden="true"></span><input value="" id="headlessui-combobox-input-1" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="w-full placeholder-gray-400 dark:placeholder-gray-500 bg-transparent border-0 text-gray-900 dark:text-white focus:ring-0 focus:outline-none sm:text-sm h-12 px-4 ps-11" placeholder="Search..." aria-label="Search..." autocomplete="off">
    <!--v-if-->
  </div>
  <div class="flex flex-col items-center justify-center flex-1 px-6 py-14 sm:px-14"><span class="i-heroicons-magnifying-glass-20-solid w-6 h-6 mx-auto text-gray-400 dark:text-gray-500 mb-4" aria-hidden="true"></span>
    <p class="text-sm text-center text-gray-900 dark:text-white">We couldn't find any items with that term. Please try again.</p>
  </div>
</div>"
`;

exports[`CommandPalette > renders with nullable correctly 1`] = `
"<div class="flex flex-col flex-1 min-h-0 divide-y divide-gray-100 dark:divide-gray-800" data-headlessui-state="">
  <div class="relative flex items-center"><span class="i-heroicons-magnifying-glass-20-solid pointer-events-none absolute start-4 text-gray-400 dark:text-gray-500 h-5 w-5" aria-hidden="true"></span><input value="" id="headlessui-combobox-input-2" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="w-full placeholder-gray-400 dark:placeholder-gray-500 bg-transparent border-0 text-gray-900 dark:text-white focus:ring-0 focus:outline-none sm:text-sm h-12 px-4 ps-11" placeholder="Search..." aria-label="Search..." autocomplete="off">
    <!--v-if-->
  </div>
  <div class="flex flex-col items-center justify-center flex-1 px-6 py-14 sm:px-14"><span class="i-heroicons-magnifying-glass-20-solid w-6 h-6 mx-auto text-gray-400 dark:text-gray-500 mb-4" aria-hidden="true"></span>
    <p class="text-sm text-center text-gray-900 dark:text-white">We couldn't find any items with that term. Please try again.</p>
  </div>
</div>"
`;

exports[`CommandPalette > renders with placeholder correctly 1`] = `
"<div class="flex flex-col flex-1 min-h-0 divide-y divide-gray-100 dark:divide-gray-800" data-headlessui-state="">
  <div class="relative flex items-center"><span class="i-heroicons-magnifying-glass-20-solid pointer-events-none absolute start-4 text-gray-400 dark:text-gray-500 h-5 w-5" aria-hidden="true"></span><input value="" id="headlessui-combobox-input-0" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="w-full placeholder-gray-400 dark:placeholder-gray-500 bg-transparent border-0 text-gray-900 dark:text-white focus:ring-0 focus:outline-none sm:text-sm h-12 px-4 ps-11" placeholder="Emptystate placeholder" aria-label="Emptystate placeholder" autocomplete="off">
    <!--v-if-->
  </div>
  <div class="flex flex-col items-center justify-center flex-1 px-6 py-14 sm:px-14"><span class="i-heroicons-magnifying-glass-20-solid w-6 h-6 mx-auto text-gray-400 dark:text-gray-500 mb-4" aria-hidden="true"></span>
    <p class="text-sm text-center text-gray-900 dark:text-white">We couldn't find any items with that term. Please try again.</p>
  </div>
</div>"
`;

exports[`CommandPalette > renders with searchable correctly 1`] = `
"<div class="flex flex-col flex-1 min-h-0 divide-y divide-gray-100 dark:divide-gray-800" data-headlessui-state="">
  <div class="relative flex items-center" style="display: none;"><span class="i-heroicons-academic-cap pointer-events-none absolute start-4 text-gray-400 dark:text-gray-500 h-5 w-5" aria-hidden="true"></span><input value="" id="headlessui-combobox-input-3" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="w-full placeholder-gray-400 dark:placeholder-gray-500 bg-transparent border-0 text-gray-900 dark:text-white focus:ring-0 focus:outline-none sm:text-sm h-12 px-4 ps-11" placeholder="Search..." aria-label="Search..." autocomplete="off">
    <!--v-if-->
  </div>
  <div class="flex flex-col items-center justify-center flex-1 px-6 py-14 sm:px-14"><span class="i-heroicons-magnifying-glass-20-solid w-6 h-6 mx-auto text-gray-400 dark:text-gray-500 mb-4" aria-hidden="true"></span>
    <p class="text-sm text-center text-gray-900 dark:text-white">We couldn't find any items with that term. Please try again.</p>
  </div>
</div>"
`;

exports[`CommandPalette > renders with selected icon correctly 1`] = `
"<div class="flex flex-col flex-1 min-h-0 divide-y divide-gray-100 dark:divide-gray-800" data-headlessui-state="">
  <div class="relative flex items-center"><span class="i-heroicons-magnifying-glass-20-solid pointer-events-none absolute start-4 text-gray-400 dark:text-gray-500 h-5 w-5" aria-hidden="true"></span><input value="" id="headlessui-combobox-input-11" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="w-full placeholder-gray-400 dark:placeholder-gray-500 bg-transparent border-0 text-gray-900 dark:text-white focus:ring-0 focus:outline-none sm:text-sm h-12 px-4 ps-11" placeholder="Search..." aria-label="Search..." autocomplete="off" aria-controls="headlessui-combobox-options-12" aria-activedescendant="headlessui-combobox-option-13">
    <!--v-if-->
  </div>
  <div id="headlessui-combobox-options-12" role="listbox" data-headlessui-state="" aria-label="Commands" class="relative flex-1 overflow-y-auto divide-y divide-gray-100 dark:divide-gray-800 scroll-py-2">
    <div class="p-2" role="option">
      <h2 class="px-2.5 my-2 text-xs font-semibold text-gray-900 dark:text-white">Group 1</h2>
      <div class="text-sm text-gray-700 dark:text-gray-200" role="listbox" aria-label="Group 1">
        <div id="headlessui-combobox-option-13" role="option" aria-selected="true" class="flex justify-between select-none items-center rounded-md px-2.5 py-1.5 gap-2 relative bg-gray-100 dark:bg-gray-800 text-gray-900 dark:text-white cursor-pointer" data-headlessui-state="active selected">
          <div class="flex items-center gap-1.5 min-w-0">
            <!--v-if-->
            <div class="flex items-center gap-1.5 min-w-0">
              <!--v-if--><span class="truncate">Command 1</span><!-- eslint-disable-next-line vue/no-v-html -->
              <!--v-if-->
            </div>
          </div><span class="i-heroicons-check h-5 w-5 text-gray-900 dark:text-white flex-shrink-0" aria-hidden="true"></span>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
